### PR TITLE
Fix casing in import statement

### LINKED ISSUE #38550: 'Nvveen/Gotty' import casing statement
**Description**

The import statement in pkg/jsonmessage/jsonmessage.go has some mixed casing (more or less the same issue as #33641). When using Bazel/Gazelle and including this as a dependancy, this can result in the following errors:

> GoCompile: missing strict dependencies:
>         /home/foo/.cache/bazel/_bazel_foo/c42f0cbd326bce63fb67245d6d8e7267/sandbox/processwrapper-sandbox/9/execroot/__main__/external/com_github_docker_docker/pkg/jsonmessage/jsonmessage.go: import of "github.com/Nvveen/Gotty"
> Known dependencies are:
>         github.com/docker/docker/pkg/term
>         github.com/docker/go-units
>         github.com/nvveen/gotty

**Additional information you deem important (e.g. issue happens only occasionally):**

I wonder how important this is considering that there is also a PR open to replace this specific library with another one (see #38398).

### PR DESCRIPTION
**Pull request by mickael-carl**
Fixes #38550

### CODE REVIEW

File: vendor.conf
Comment (thaJeztah) [inline vendor.conf]: We can't change this; the repository itself is cased like this; see https://github.com/Nvveen/Gotty

There's a pull request being reviewed though that removes this dependency 


Also;  changing the name upstream would cause a lot of issues, eg see https://github.com/sirupsen/logrus/issues/451 and https://github.com/sirupsen/logrus/pull/384

**Review by thaJeztah — CHANGES_REQUESTED** [review]
I don't think we can do this; see my comment below

**Comment by mickael-carl** [comment]
That's more or less what I suspected. I will wait until #38398 is merged then!

File: vendor.conf
Comment (SvenDowideit) [reply vendor.conf]: it still _is_ causing issues - and in new code:

I'm getting 

```
go: github.com/Sirupsen/logrus@v1.3.0: parsing go.mod: unexpected module path "github.com/sirupsen/logrus"
--
```

from trying to use `go mod` with docker/engine and docker/cli

Comment (thaJeztah) [reply vendor.conf]: @SvenDowideit which repository are you fetching from? I think `go mod` (when pointing at _this_ repository) picks up the latest _tag_ (not master), so you'd get a really old codebase (17.05); does it work if you point it at the https://github.com/docker/engine repository? (that should get the latest 17.09 codebase)

Comment (SvenDowideit) [reply vendor.conf]: its become very weird and frustrating, and so i need to make a simple example, but the Sirupson one is when I _only_ import "github.com/docker/engine" and try to get go mod to use v18.09.1
somewhere in the deps I have it ends up adding "github.com/docker/docker" and then things go boom.

**Pull request closed without merging**
